inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/service/RecommendationServiceImpl.java

```diff
@@ -1009,7 +1010,7 @@ public Predictions computePredictions(User aUser, Project aProject,
 
                         predictions.putPredictions(layer.getId(), suggestions);
                     }
-                    catch (Throwable e) {
+                    catch (RecommendationException | UIMAException | IOException e) {
```

Comment:
The reason why a Throwable is used instead of specific exceptions is to prevent that an implementation error in a recommender breaks the service, i.e. prevents other recommenders to be run. I'd say that is intended.

Reply:
It would probably be a good idea to add a comment here (and in similar places) though saying that it is in fact intentional.

Reply:
catching Throwable is intentional here as we want to continue the execution even if a particular recommender fails.

Reply:
:) No, what I meant was adding a comment in the code... The comment must make sense, so adapt as needed.

Reply:
Sorry, teacher. I will submit tomorrow